src/Compliance.Plugins/PALegalConsultationPlugin.cs

```diff
@@ -21,61 +21,101 @@ public PALegalConsultationPlugin()
+
+            if (pluginExecutionContext.InputParameters.Contains(InputParameter.Target) && pluginExecutionContext.InputParameters[InputParameter.Target] is Entity modifiedPAFile)
+            {
+                // Get the original record
+                var originalPAFile = pluginExecutionContext.PreEntityImages["PreEntityImage"];
+
+                if (originalPAFile is null || modifiedPAFile is null)
+                {
+                    localContext.Trace("No modified or original record, exiting.");
+                    return;
+                }
+
+                localContext.Trace("Update was triggered by PALegalConsultation plugin. Adding back original 'Modified On' value.");
+
+                // Reset modified on
+                modifiedPAFile["modifiedon"] = originalPAFile.Attributes["modifiedon"];
+            }
+        }
+
+        private void UpdateLegalConsultationFields(LocalPluginContext localContext)
+        {
+            var organizationService = localContext.OrganizationService;
+            var pluginExecutionContext = localContext.PluginExecutionContext;
+
+            // Prevent infinite loops
+            if (pluginExecutionContext.Depth > 1)
+            {
+                return;
+            }
+
+            // The InputParameters collection contains all the data passed in the message request.
+            if (pluginExecutionContext.InputParameters.Contains(InputParameter.Target) && pluginExecutionContext.InputParameters[InputParameter.Target] is EntityReference entityRef)
+            {
+                // Obtain the target entity from the input parameters.
+                var paFile = organizationService.Retrieve<opc_pafile>(
+                    entityRef,
+                    EntityHelper.GetAttributeLogicalName<opc_pafile>(x => x.opc_legalconsultationsearcherurl),
+                    EntityHelper.GetAttributeLogicalName<opc_pafile>(x => x.opc_LegalConsultations)
+                );
+
+                if (paFile != null)
+                {
+                    // Retrieve the list of legal consultations associated to this PA file
+                    var entityGuid = pluginExecutionContext.PrimaryEntityId;
+                    var expression = new QueryExpression(opc_legalconsultation.EntityLogicalName)
+                    {
+                        ColumnSet = new ColumnSet(opc_legalconsultation.PrimaryNameAttribute)
+                    };
+                    expression.Criteria.AddCondition(EntityHelper.GetAttributeLogicalName<opc_legalconsultation>(x => x.opc_PAFileId), ConditionOperator.Equal, entityGuid);
+                    var legalConsultationNumbers = organizationService.RetrieveMultiple<opc_legalconsultation>(expression)
+                        .Select(x => x.opc_name)
+                        .OrderBy(x => x);
+
+                    var legalConsultationNumbersJoined = legalConsultationNumbers.Count() > 0 ? string.Join(", ", legalConsultationNumbers) : null;

```

Comment:
This could be refactored to the following:

```suggestion
                    var legalConsultationNumbersJoined = string.Join(", ", legalConsultationNumbers);
```